js/stack/stack-data-structure.js

```diff
@@ -5,22 +5,36 @@ class StackDataStructure {
   }
 
   canPush() {
-    // ... your code goes here
+    if (this.stackControl.length === this.MAX_SIZE) {
+      return false
+    }
+    return true
   }
 
   display() {
     // ... your code goes here
   }
 
   isEmpty() {
-    // ... your code goes here
+    if (this.stackControl.length === 0) {
+      return true
+    }
+    return false
   }
 
   push(item) {
-    // ... your code goes here
+    if (!this.canPush()) {
+      return 'Stack Overflow'
+    }
+    this.stackControl.push(item)
+    return this.stackControl
   }
 
   pop() {
-    // ... your code goes here
+    if (this.stackControl.length === 0) {
+      return 'Stack Underflow'
+    } 
+    this.stackControl.pop()
+    return this.stackControl[this.stackControl.length - 1]
```

Comment:
La forma idonea de abordar esto era:
```suggestion
    if (this.stackControl.length === 0) {
      return 'Stack Underflow'
    } 
   
    return this.stackControl.pop()
```

array.pop() retorna el elemento eliminado :smile: